### DISCUSSION TITLE
chnkserver refuses to stop - how can I get more information?

### DISCUSSION THREAD

**Discussion by BillKen** [moosefs/moosefs#384] [Q&A]
## Have you read through available documentation and open Github issues?

Yes

## Is this a BUG report, FEATURE request, or a QUESTION? Who is the indended audience?

QUESTION for the COMMUNITY

#### Your moosefs version and its origin (moosefs.com, packaged by distro, built from source, ...).
3.0.114, build 1257 from git. (compiled from source)

#### Operating system (distribution) and kernel version.

gentoo 32bit arm

#### Hardware / network configuration, and underlying filesystems on master, chunkservers, and clients.

32bit arm on odroid HC2, xfs on a 6TB seagate ironwolf

#### How much data is tracked by moosefs master (order of magnitude)?

 - All fs objects: 103218637
 - Total space: 22 TiB	
 - Free space: 7.0 TiB
 - RAM used: 24 GiB	
 - last metadata save duration: ~1.8m

## Describe the problem you observed.
An error is shown on a disk in the web interface with no corresponding log or system entry in dmesg or system logs.  Its usuallyafter a crash (power problems).  The chunkserver refuses to shutdown and has to be killed.  It will restart and run normally but web interface will show an error (with no information as to what causes it) after a few minutes and it will then no longer shut shut down.  The only cure I have found so far is to reformat the disk and wait for it to backfill.  Normal shutdown time is 10-15 seconds so waiting 1800 seconds is excessive.

#### Can you reproduce it? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.
Occurs randomly after a crash

#### Include any warning/errors/backtraces from the system logs

```Sep 20 17:58:11 san0 mfschunkserver[13046]: about 1800 seconds passed and lockfile is still locked - giving up```

**Comment by chogata** [comment]
There is an issue currently in MooseFS, that causes the CS to be impossible to stop via normal means (sigstop) if too many I/O errors occur on a drive where chunks are kept. It will be fixed in version 3.0.115

BTW, the errors on disks interface are I/O errors, that means MooseFS chunk server tried to read and/or modify data on a local hard drive and failed (got I/O error from kernel). I don't know why nothing shows in your system log, because it should - are you sure your OS reports I/O errors correctly?

**Comment by BillKen** [comment]
More information:
```
Sep 20 21:43:57 mfsmaster mfsmaster[3533]: chunk 00000000009C09D9_00000001: chunk hasn't been deleted since previous loop - retry
Sep 20 21:51:02 mfsmaster mfsmaster[3533]: chunk 00000000009C09D9_00000001: chunk hasn't been deleted since previous loop - retry
Sep 21 06:42:18 mfsmaster mfsmaster[3521]: chunk 00000000009C09D9 has only copies with wrong versions (1) - please repair it manually
Sep 21 06:42:18 mfsmaster mfsmaster[3521]: chunk 00000000009C09D9_00000001 - copy with wrong version on (192.168.72.2 - ver:00000000)
Sep 21 06:47:59 mfsmaster mfsmaster[3521]: chunk 00000000009C09D9 has only copies with wrong versions (1) - please repair it manually
```
As the chunk should have been deleted, I decided to check my backups are up to date and manually delete all the chunks (goal of 4, should have copied them off, but I have backups :) as the log says they were not deleted.  The error message then changed to "wrong version" with a 00000000 version.  Searching the disk found a zero byte file on the problematic chunkserver (other chunks were on good chunkservers shut down normally.)

1. the error message masked the real problem by not listing the wrong version - could it be more informative?  It only changed when the cluster was rebooted from cold this morning.
2. is there a way of creating a valid blank chunk?  mfschunktool fails to read the header of a file and exits. (either a zero byte file or some random data - I am not enthused about copying another random chunk and renaming it ... or is that practical?

**Comment by BillKen** [comment]
Hi, yes it is showing hardware errors when they occur.  I presume that the system did not get a chance to write to a log in this case as it was a power failure on the chunkservers (common +12v supply)

After restarting the chunkservers an error may appear against a disk (usually a minute or two later) in the CGI interface.  There is nothing in any log that corresponds with that error message at that time.  At some time later various error messages may appear (or may not) - the one above is the only one I have been able to track down so far.

Is there a way to see what caused the error to be listed in the CGI?  - waiting for a check loop to show something can take a long time.

**Comment by chogata** [comment]
I do not understand why you would like to create an empty chunk. Is it because you want to restore a missing file via filling the missing parts with zeros? If yes, then `mfsfilerepair` will do it for you automatically, no need to tamper with separate chunks.

You can see the chunk ID in CGI - just hover over the error date and you will see the chunk ID that was accessed and caused the i/o error.
